### PR TITLE
Add Python 3.10 builds.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Enables cibuildwheel for Python 3.10.

~Currently CI is failing because there aren't wheels available for scipy yet. Building scipy from source fails. This should be re-tried in a few weeks once scipy wheels are available for Python 3.10.~ edit: scipy 1.7.2 is now released, so wheels are available and freud's CI passes.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ and this project adheres to
 
 ### Added
 * `freud.diffraction.StaticStructureFactorDirect` class (unstable) can be used to compute the static structure factor S(k) by sampling reciprocal space vectors.
+* Python 3.10 is supported.
+
```
Comment (bdice) [inline ChangeLog.md:12]:
```suggestion
```

**Comment by bdice** [comment]
Still seeing some issues with getting scipy wheels here. It looks like the failure to install scipy from wheels or from source also prevents other testing dependencies from being installed properly (matplotlib, garnett, gsd, sympy), resulting in failed tests.

**Comment by bdice** [comment]
The new version of scipy works, so this is ready to review and merge.

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
We need to actually be running the unit tests in a python 3.10 environment, not just confirming that we can build freud against python 3.10.

**Comment by bdice** [comment]
@tommy-waltmann cibuildwheel also runs all unit tests. https://github.com/glotzerlab/freud/blob/b0110550a8bba529c422051b24340c49eeb6e073/.github/workflows/build_wheels.yml#L57-L61

I would treat cibuildwheel as our "primary" CI at this point. We've only kept the other CircleCI jobs for two reasons, in my recollection:
- We don't have Windows tests on cibuildwheel yet. See stalled PR #749.
- We only run the "style check" job with cppcheck and clang-tidy on CircleCI (not really a job for cibuildwheel, and it's too complicated for pre-commit CI because clang-tidy requires building the package).

I believe the other CircleCI tests for Linux are redundant with those in cibuildwheel (aside from minor differences in the underlying CI images like compiler versions and TBB versions, but we don't really have those documented in our test matrix).

**Comment by tommy-waltmann** [comment]
Then we should still add CircleCI tests for python 3.10 on windows in this PR before merging. Later we can decide what to do with the rest of the CircleCI tests in another issue or PR.

**Comment by bdice** [comment]
> Then we should still add CircleCI tests for python 3.10 on windows in this PR before merging. Later we can decide what to do with the rest of the CircleCI tests in another issue or PR.

Good idea. I added a Windows Python 3.10 build to CircleCI. Please take a look once CI passes.

**Review by tommy-waltmann — APPROVED** [review]
Looks good, I'll make an issue for the CircleCI tests which are redundant.

**Pull request merged**